src/java.desktop/share/classes/sun/java2d/pipe/RenderBuffer.java

```diff
@@ -170,9 +170,9 @@ public RenderBuffer put(short[] x) {
     public RenderBuffer put(short[] x, int offset, int length) {
         // assert (position() % SIZEOF_SHORT == 0);
         if (length > COPY_FROM_ARRAY_THRESHOLD) {
-            long offsetInBytes = offset * SIZEOF_SHORT + Unsafe.ARRAY_SHORT_BASE_OFFSET;
-            long lengthInBytes = length * SIZEOF_SHORT;
-            unsafe.copyMemory(x, offsetInBytes, null, curAddress, lengthInBytes);
+            int offsetInBytes = offset * SIZEOF_SHORT;
+            int lengthInBytes = length * SIZEOF_SHORT;
+            MemorySegment.copy(x, offsetInBytes, segment, JAVA_SHORT, curOffset, length);
             position(position() + lengthInBytes);
```

Comment:
If you want, you could use `MemoryLayout::scale` here, i.e.
```suggestion
            position(JAVA_SHORT.scale(position(), length));
```
(up to preference though)